raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -1004,32 +1003,28 @@ private boolean handleFetchResponse(
 
         FollowerState state = quorum.followerStateOrThrow();
         if (error == Errors.NONE) {
-            FetchResponseData.OffsetAndEpoch nextOffsetAndEpoch = partitionResponse.nextOffsetAndEpoch();
-            if (nextOffsetAndEpoch.nextFetchOffset() > 0) {
+            long truncationOffset = partitionResponse.truncationOffset();
+            if (truncationOffset >= 0) {
                 // The leader is asking us to truncate before continuing
-                OffsetAndEpoch nextFetchOffsetAndEpoch = new OffsetAndEpoch(
-                    nextOffsetAndEpoch.nextFetchOffset(), nextOffsetAndEpoch.nextFetchOffsetEpoch());
-
                 state.highWatermark().ifPresent(highWatermark -> {
-                    if (nextFetchOffsetAndEpoch.offset < highWatermark.offset) {
-                        throw new KafkaException("The leader requested truncation to offset " +
-                            nextFetchOffsetAndEpoch.offset + ", which is below the current high watermark" +
-                            " " + highWatermark);
+                    if (truncationOffset < highWatermark.offset) {
```

Comment:
@hachikuji can you check this comparison? Should it be `<=`? Is the `highWatermark` the offset that is committed or is it the offset greater than the committed offset?